runelite-client/src/main/java/net/runelite/client/plugins/StatusBars/StatusBarsOverlay.java

```diff
@@ -0,0 +1,305 @@
+	}
+
+	@Override
+	public Dimension render(Graphics2D g)
+	{
+		// gets the current viewport and changes values accordingly so the bars are in the right position.
+		for (Viewport viewport : Viewport.values())
+		{
+			Widget viewportWidget = client.getWidget(viewport.getViewport());
+
+			if (viewportWidget != null && !viewportWidget.isHidden())
+			{
+				Widget widget = client.getWidget(viewport.getViewport());
+				Point location = viewportWidget.getCanvasLocation();
+				Point offsetLeft = viewport.getOffsetLeft();
+				Point offsetRight = viewport.getOffsetRight();
+				int height = HEIGHT;
+				int offsetPrayerX = (location.getX() - offsetRight.getX());
+				int offsetHealthX = (location.getX() - offsetLeft.getX());
+				int offsetHealthY = (location.getY() - offsetLeft.getY());
+				int offsetPrayerY = (location.getY() - offsetRight.getY());
+				Widget widgetBankTitleBar = client.getWidget(WidgetInfo.BANK_TITLE_BAR);
+
+				if (viewport == Viewport.RESIZED_BOTTOM && !widget.isHidden())
+				{
+					offsetPrayerY = (location.getY() - 18 - offsetRight.getY());
+					offsetHealthY = (location.getY() - 18 - offsetRight.getY());
+					height += 20;
+				}
+				else if (viewport == Viewport.RESIZED_BOX && widget.isHidden())
+				{
+					return null;
+				}
+				else if (widgetBankTitleBar != null && !widgetBankTitleBar.isHidden())
+				{
+					return null;
+				}
+				else
+				{
+					offsetPrayerX = (location.getX() + widget.getWidth() - offsetRight.getX());
+				}
+
+				// Get Max HP levels and current HP levels
+				final int maxHealth = client.getRealSkillLevel(Skill.HITPOINTS);
+				final int maxPrayer = client.getRealSkillLevel(Skill.PRAYER);
+				final int currentHealth = client.getBoostedSkillLevel(Skill.HITPOINTS);
+				final int currentPrayer = client.getBoostedSkillLevel(Skill.PRAYER);
+
+				// Check if player = poisoned or venomed and change color.
+				Color healthBar;
+				final int poisonState = client.getVar(VarPlayer.IS_POISONED);
+				if (poisonState > 0 && poisonState < 37)
+				{
+					healthBar = POISONED_COLOR;
+				}
+				else if (poisonState > 1000000)
+				{
+					healthBar = VENOMED_COLOR;
+				}
+				else
+				{
+					healthBar = HEALTH_COLOR;
+				}
+
+				// Checks if the player has activated quick prayers
+				final int quickPrayerState = client.getVar(Varbits.QUICK_PRAYER);
+				final Color prayerBar = quickPrayerState == 1 ? QUICK_PRAYER_COLOR : PRAYER_COLOR;
+
+				// Render the health and prayer bars
+				renderBar(g, offsetHealthX , offsetHealthY,
+					maxHealth, currentHealth, height, healthBar);
+
+				renderBar(g, offsetPrayerX, offsetPrayerY ,
+					maxPrayer, currentPrayer, height, prayerBar);
+
+				// If the player hovers over a food or prayer restoring item this
+				// for loop will get the values from the item stats plugin
+				int prayerHealValue = 0;
+				int foodHealValue = 0;
+				WidgetItem[] inventory = queryRunner.runQuery(new InventoryWidgetItemQuery());
```

Comment:
This entire widget query for all inventory items per frame seems sub-optimal

Reply:
What would be a better solution for this? this is how itemstats plugin also does it